security/tls-authorization/README.adoc

```diff
@@ -0,0 +1,268 @@
+* `$ mvn clean package`
+. Build the example certificates and keystores with the following scripts or directly with the Java keytool commands:
+* `create_unsigned_certificates.sh`
+* `create_signed_certificates.sh`
++
+These scripts create keystores and trust stores in the project directory.
+. Download at least one Infinispan Server instance.
+
+== Unsigned Server certificates
+
+. Copy `server-keystore.pfx` to `$ISPN_HOME/server/conf`.
+. Copy `server/infinispan-uCert.xml` to `$ISPN_HOME/server/conf`.
+. Use the `infinispan-uCert.xml` configuration, changes are
++
+Note that the default properties realm is removed.
++
+[source,xml]
+----
+<security-realm name="default">
+  <server-identities>
+    <ssl>
+      <keystore path="server-keystore.pfx"
+                keystore-password="secret"
+                alias="infinispan-server"/>
+    </ssl>
+  </server-identities>
+</security-realm>
+----
++
+[source,xml]
+----
+<distributed-cache name="secured" mode="SYNC">
+  <encoding media-type="application/x-protostream"/>
+</distributed-cache>
+----
++
+. Start the server.
++
+[source]
+----
+$ `$ISPN_HOME/bin/server.sh -c infinispan-uCert.xml`
+----
++
+. Run the client.
++
+[source]
+----
+$ export TEST=SIMPLE && mvn exec:exec@run
+----
+. Check that the put and get succeeds.
+
+=== Unsigned Server certificates with client authorization
+
+. Copy `client-truststore.pfx` to `$ISPN_HOME/server/conf`.
+. Copy `server/infinispan-uCertAuth.xml` to `$ISPN_HOME/server/conf`.
+. Use the `infinispan-uCertAuth.xml` configuration, changes are
++
+[source,xml]
+----
+<endpoints socket-binding="default"
+           security-realm="default"
+           require-ssl-client-auth="true"/>
+----
++
+[source,xml]
+----
+<security-realm name="default">
+  <server-identities>
+    <ssl>
+      <keystore path="server-keystore.pfx"
+                keystore-password="secret"
+                alias="infinispan-server"/>
+      <truststore path="client-truststore.pfx"
+                  password="ClientSecret"/>
+    </ssl>
+  </server-identities>
+  <truststore-realm/>
+</security-realm>
+----
++
+[source,xml]
+----
+<cache-container>
+  <security>
+    <authorization>
+      <common-name-role-mapper/>
+      <!-- In this example `Client1` and `Client2` correspond to the `cn` field in client certificates -->
+      <role name="Client1" permissions="ALL"/>
+      <role name="Client2" permissions="READ"/>
+    </authorization>
+  </security>
+</cache-container>
+----
++
+[source,xml]
+----
+<distributed-cache name="secured" mode="SYNC">
+  <encoding media-type="application/x-protostream"/>
+  <security>
+    <authorization/>
+  <security>
+</distributed-cache>
+----
++
+. Start the server.
++
+[source]
+----
+$ `$ISPN_HOME/bin/server.sh -c infinispan-uCertAuth.xml`
+----
++
+. Run the client.
++
+[source]
+----
+$ export TEST=SIMPLEAUTH && mvn exec:exec@run
+----
+. Check that put and get succeeds both times as Client1 and Client2 certificates are used.
+
```

Comment:
Got this exception in logs: 

ISPN005003: Exception reported java.lang.SecurityException: ISPN000287: Unauthorized access: subject 'Subject with principal(s): [anonymous, CN=Client2, OU=Infinispan, O=JBoss, L=Red Hat, InetAddressPrincipal [address=127.0.0.1/127.0.0.1]]' lacks 'WRITE' permission

If this is expected, we should explain to users.